Flow.Launcher.Plugin/SharedCommands/SearchWeb.cs

```diff
@@ -1,18 +1,45 @@
+using Microsoft.Win32;
 using System;
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
 
+
 namespace Flow.Launcher.Plugin.SharedCommands
 {
     public static class SearchWeb
     {
+        private static string GetDefaultBrowserPath()
+        {
+            string name = string.Empty;
+            try
+            {
+                using var regDefault = Registry.CurrentUser.OpenSubKey("Software\\Microsoft\\Windows\\Shell\\Associations\\UrlAssociations\\http\\UserChoice", false);
+                var stringDefault = regDefault.GetValue("ProgId");
+
+                using var regKey = Registry.ClassesRoot.OpenSubKey(stringDefault + "\\shell\\open\\command", false);
+                name = regKey.GetValue(null).ToString().ToLower().Replace("\"", "");
+
+                if (!name.EndsWith("exe"))
+                    name = name.Substring(0, name.LastIndexOf(".exe") + 4);
+
+            }
+            catch
+            {
+                return string.Empty;
+            }
+
+            return name;
+        }
+
         /// <summary> 
         /// Opens search in a new browser. If no browser path is passed in then Chrome is used. 
         /// Leave browser path blank to use Chrome.
         /// </summary>
 		public static void NewBrowserWindow(this string url, string browserPath = "")
         {
+            browserPath = string.IsNullOrEmpty(browserPath) ? GetDefaultBrowserPath() : browserPath;
```

Comment:
with getting the default browser from user, which could be any browser right, do we know if it will support opening in new window?

Reply:
It works on firefox and chromium-based browsers, and there is a fix for IE a few lines later.... I think this covers pretty much every browser.